### PR TITLE
New source: Arabs Hentai (ar)

### LINKED ISSUE #19: ArabsHentai : Source request
### Source name

ArebsHentai 

### Source link

https://arabshentai.com/

### Source language

Arabic

### Other details

NSFW : Yes

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/keiyoushi/extensions-source/) and verified it does not appear in the code base.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by cuong-tran**
Close: #19

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/page/$page/?s=$query".toHttpUrl().newBuilder()
```
Comment (vetleledaal) [inline src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt]: You need to escape the query properly

File: src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) =
+        SManga.create().apply {
+            document.selectFirst(".content")?.run {
+                selectFirst(".sheader .poster img")?.run {
+                    thumbnail_url = imgAttr()
+                }
+                selectFirst(".sheader .data h1")?.run {
+                    title = text()
+                }
```
Comment (AwkwardPeak7) [inline src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt]:
```suggestion
            document.selectFirst(".content")!!.run {
                thumbnail_url = selectFirst(".sheader .poster img")?.imgAttr()
                title = selectFirst(".sheader .data h1")!!.text()
```

File: src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    private val otherNameSelector =
+        "div b:contains(أسماء أُخرى) + span"
+    private val seriesStatusSelector =
+        "div b:contains(حالة المانجا) + span"
+    private val seriesArtistSelector =
+        "div b:contains(الرسام) + span a"
+    private val seriesAuthorSelector =
+        "div b:contains(الكاتب) + span a"
+    private val seriesTypeSelector =
+        "div b:contains(نوع العمل) + span a"
```
Comment (AwkwardPeak7) [inline src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt]: inline these

File: src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+            }
+        }
+
+    private fun String?.parseChapterDate(): Long {
+        if (this == null) return 0
+        return try {
+            dateFormat.parse(this)?.time ?: 0
+        } catch (_: Exception) {
+            0
+        }
```
Comment (AwkwardPeak7) [inline src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt]:
```suggestion
        return try {
            dateFormat.parse(this)!!.time
        } catch (_: ParseException) {
            0L
        }
```

File: src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    private var fetchGenresAttempts: Int = 0
+
+    private fun fetchGenres() {
+        if (fetchGenresAttempts < 3 && genreList.isEmpty()) {
+            try {
+                client.newCall(genresRequest()).execute()
+                    .let {
+                        val document = it.asJsoup()
+                        genreList = parseGenres(document)
+                    }
```
Comment (AwkwardPeak7) [inline src/ar/arabshentai/src/eu/kanade/tachiyomi/extension/ar/arabshentai/ArabsHentai.kt]:
```suggestion
                genreList = client.newCall(genresRequest()).execute()
                    .asJsoup()
                    .let(::parseGenres)
```

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]
rebase on main

**Pull request merged**